integration/src/integration/java/pl/allegro/tech/hermes/integration/PublishingTest.java

```diff
@@ -319,4 +329,36 @@ public void shouldPublishAndConsumeMessageWithTraceId() {
         assertThat(response).hasStatus(Response.Status.CREATED);
         assertThat(remoteService.waitAndGetLastRequest()).hasHeaderValue("Trace-Id", traceId);
     }
+
+    @Test
+    public void shouldPublishAndConsumeMessageWithTraceAndSpanHeaders() {
+
+        // given
+        String message = "{\"id\": 101}";
+        TraceContext trace = new TraceContext(
+                UUID.randomUUID().toString(),
+                UUID.randomUUID().toString(),
+                UUID.randomUUID().toString(),
+                "1",
+                "0"
+        );
```

Comment:
...but my point was to actually have less text to read, especially if values are random or arbitrary :-)

For instance something like this would be more concise but still readable:
```
// given
TraceHeaders headers = TraceHeaders.random();

// when
Response response = headers.addAll(client.request()).post()

// then
assertThat(response.headers().entrySet()).containsAll(headers.entrySet())
```